js/selectors/activity-feedback-selectors.js

```diff
@@ -0,0 +1,293 @@
+import { createSelector } from 'reselect'
+import { fromJS, Map as IMap } from 'immutable'
+import {
+  AUTOMATIC_SCORE,
+  MAX_SCORE_DEFAULT,
+  RUBRIC_SCORE
+} from '../util/scoring-constants'
+
+/******************************************************************
+ * These functions compute data for views. Some of the computations
+ * are expensive, so we use reselect to memoize the results. Beacause
+ * we use the selectors on multiple containers within the same
+ * document we use the `makeGet<property>` factory method pattern
+ * to bind each memoized function with the appropriate container.
```

Comment:
I've been using reselect and I've never had to do this additional memoization and `make...` pattern. I guess the reason to do that is not really because we have multiple containers, but because some of the selectors are based on component properties (e.g. activity), right?

If we use just the global, singular Redux state, then obviously result will be the same for every component, so we don't need to worry about the memoization. But if also use component-specific properties, then we do need separate instances for every component. Is my logic correct? As I'm saying, I've never seen this pattern before, so I'm wondering if I understand it correctly.

If so, maybe we could update comments here that this is necessary only if your selector is based on properties. And review selectors if they really need to use this pattern? Or are all of them based on properties? It seems that only `getActivity` is property-based, but also it's quite likely that this activity is passed almost everywhere.

Also, I think I'd find it helpful to group functions into three categories (perhaps just using some comments and order):
- helpers
- simple state-based inputs (selectors)
- complex selectors using other selectors as inputs

I think this is almost like this now, but maybe some clear division could be nice.

I've noticed that sometimes we have only `makeGetSomething` definition, and sometimes we additionally specify `getSomething` as a separate function. I'd lean towards just single `makeGetSomething` definition, so I can clearly see that this code is a selector. Or at least keep them next to each other (I think that's always the case except from `getQuestions`).